src/main.js

```diff
@@ -3,15 +3,32 @@ import router from './services/router'
 import App from './App'
 import store from './vuex/store'
 import { sync } from 'vuex-router-sync'
-import { loginFromCookie } from './vuex/modules/auth/actions'
+import { loginFromCookie, checkFirstAdmin } from './vuex/modules/auth/actions'
+import { isConnected } from './services/kuzzleWrapper'
+import Promise from 'bluebird'
 
 Vue.config.debug = process.env.NODE_ENV !== 'production'
 
 sync(store, router)
 
-loginFromCookie(store, () => {
-  router.start({
-    store,
-    components: { App }
-  }, 'body')
-})
+isConnected()
+  .then(() => {
+    return loginFromCookie(store)
+  })
+  .then((user) => {
+    if (!user) {
+      return checkFirstAdmin(store)
+    }
+
+    return Promise.resolve()
+  })
+  .then(() => {
+    router.start({
+      store,
+      components: { App }
+    }, 'body')
+  })
+  .catch((err) => {
+    /* TODO: do something with this... Kuzzle is not responding, what we have to do?? */
```

Comment:
IMHO something is being done wrong here. Roughly, we are executing some logic *before the application starts*, which seems fair. This logic checks whether Kuzzle Server is up or not, whether the user cookie exists or not, whether the token contained in this cookie exists or not, whether admins and users have been properly initialized or not. Fair.
But I think that the app should be able to be alive independently of these factors. IMHO
* the application boots (`router.start` is called)
* some logic checks the availability of Kuzzle Server. If Kuzzle is down, a nice error message is shown. Stop here.
* some logic checks the user cookie, token, firstAdmin, etc... Display relevant loaders and screens in reaction to the state.

Sounds ok?